dts: nordic: Add Channel Sounding support to nrf-radio

ivaniushkov opened the pull request:
- Added `cs-supported` property to nrf-radio devicetree
- Added `HAS_HW_NRF_RADIO_CS` Kconfig option which is set if `cs-supported` property is enabled
- Enabled `cs-supported` property for nrf54-series devices

eriksandgren commented on dts/bindings/net/wireless/nordic,nrf-radio.yaml:
```diff
@@ -227,3 +227,10 @@ properties:
+  cs-supported:
+    type: boolean
+    description: |
+      If set, the radio hardware supports the Channel Sounding.
```
```suggestion
      If set, the radio hardware supports the BLE Channel Sounding feature.
```

ryanjh commented on dts/common/nordic/nrf54l15.dtsi:
```diff
@@ -212,6 +212,7 @@
 				ieee802154-supported;
 				ble-2mbps-supported;
 				ble-coded-phy-supported;
+				cs-supported;
```
You might need to delete this property for nrf54l15bsim.
https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/native/nrf_bsim/nrf54l15bsim_nrf54l15_cpuapp.dts#L101

anangl commented on soc/nordic/common/Kconfig.peripherals:
```diff
@@ -207,6 +207,9 @@ config HAS_HW_NRF_RADIO_IEEE802154
 config HAS_HW_NRF_RADIO_TX_PWR_HIGH
 	def_bool $(dt_nodelabel_bool_prop,radio,tx-high-power-supported)
 
+config HAS_HW_NRF_RADIO_CS
```
Please keep the alphabetical order here.

ivaniushkov replied:
thank you, I moved it according to alphabetical order

ivaniushkov committed:
dts: nordic: Add Channel Sounding support to nrf-radio

- Added `cs-supported` property to nrf-radio devicetree
- Added `HAS_HW_NRF_RADIO_CS` Kconfig option which is set if
`cs-supported` property is enabled
- Enabled `cs-supported` property for nrf54-series devices
- Disabled `cs-supported` on nrf54l15bsim because it is not
yet supported

Signed-off-by: Ivan Iushkov <ivan.iushkov@nordicsemi.no>

jhedberg commented on dts/common/nordic/nrf54l15.dtsi:
```diff
@@ -212,6 +212,7 @@
 				ieee802154-supported;
 				ble-2mbps-supported;
 				ble-coded-phy-supported;
+				cs-supported;
```
Since there's a dedicated `bt_hci_controller` node (as a child node of the radio node), wouldn't these kind of properties be more suitable there? What's the rationale of putting Bluetooth-specific properties in the generic radio node instead of the Bluetooth-specific HCI device node?

jhedberg replied:
Hmm, just as I wrote that I realized that the upstream `zephyr,bt-hci-ll-sw-split` hasn't yet implemented channel sounding support. That said, the same question still applies, i.e. what's the reason you wouldn't want to put these into your HCI DT node for the SD controller?
